File: two-sum/twoSum.go
func twoSum(nums []int, target int) []int {
    m:= make(map[int]int)

    for i:=0; i < len(nums) ; i++ {
        num := nums[i]
        compl := target - num
        index, found := m[compl]
        if found == true {
            return []int {index, i}
        }
        m[num] = i
    }

    return []int {math.MinInt64, math.MinInt64}
}

File: two-sum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let map = {};
  for (let i = 0; i < nums.length; i++) {
    let num = nums[i];
    let compl = target - nums[i];
    if (map[compl] || map[compl] === 0) return [map[compl], i];

    map[num] = i;
  }
};
